core/src/main/scala/kafka/server/KafkaRequestHandler.scala

```diff
@@ -386,7 +431,26 @@ class BrokerTopicMetrics(name: Option[String], configOpt: java.util.Optional[Kaf
       meter.close()
   }
 
-  def close(): Unit = metricTypeMap.values.foreach(_.close())
+  def close(): Unit = {
+    metricTypeMap.values.foreach(_.close())
+    metricGaugeTypeMap.values.foreach(_.close())
+  }
+}
+
+class BrokerTopicAggregatedMetric() {
+  private val partitionMetricValues = new ConcurrentHashMap[Int, Long]().asScala
```

Comment:
```suggestion
  private val partitionMetricValues = new ConcurrentHashMap[Int, Long]()
```

Can we keep the map in java itself?

Reply:
Sure!